WebApi/Controllers/BooksController.cs

```diff
@@ -0,0 +1,261 @@
+            }
+
+            return this.Ok(resultBook);
+        }
+
+        /// <summary>
+        /// Getting book authors
+        /// </summary>
+        /// <param name="bookId">Book id to get authors</param>
+        /// <returns>List of authors</returns>
+        [HttpGet("{bookId}/authors")]
+        public IActionResult GetBookAuthors(int bookId)
+        {
+            List<Author> authors = this.library.GetBookAuthors(bookId);
+            if (authors != null && authors.Count != 0)
+            {
+                return this.Ok(authors);
+            }
+
+            return this.NotFound("Book doesn`t exist or book doesn`t containts authors!");
+        }
+
+        /// <summary>
+        /// Getting book genres
+        /// </summary>
+        /// <param name="bookId">Book id to get genres</param>
+        /// <returns>List of genres</returns>
+        [HttpGet("{bookId}/genres")]
+        public IActionResult GetBookGenres(int bookId)
+        {
+            List<Genre> genres = this.library.GetBookGenres(bookId);
+            if (genres != null && genres.Count != 0)
+            {
+                return this.Ok(genres);
+            }
+
+            return this.NotFound("Book doesn`t exist or book doesn`t containts authors!");
+        }
+
+        /// <summary>
+        /// Getting books of genre
+        /// </summary>
+        /// <param name="genreId">Genre id to get its books</param>
+        /// <returns>List of books</returns>
+        [HttpGet("{genreId}/books")]
+        public IActionResult GetBooksOfGenre(int genreId)
+        {
+            List<Book> booksOfGenre = this.library.GetBooksInGenre(genreId).ToList();
+            if (booksOfGenre.Count != 0)
+            {
+                return this.Ok(booksOfGenre);
+            }
+
+            return this.NotFound("Genre of books of genre wasn`t found!");
+        }
+
+        /// <summary>
+        /// Updating book by id
+        /// </summary>
+        /// <param name="id">Id of book to update</param>
+        /// <param name="book">Data for update</param>
+        /// <returns>Book or NotFound</returns>
+        [HttpPut("{id}")]
+        public IActionResult UpdateBook(int id, [FromBody] Book book)
+        {
+            if (this.library.UpdateBook(id, book))
+            {
+                return this.Ok(book);
+            }
+
+            return this.NotFound();
+        }
+
+        /// <summary>
+        /// Updating book genre (replacing one to other)
+        /// </summary>
+        /// <param name="bookId">Book id to update it`s genre</param>
+        /// <param name="oldGenreId">Old genre id</param>
+        /// <param name="newGenreId">New genre id</param>
+        /// <returns>Status of update</returns>
+        [HttpPut("{bookId}/genres/{oldGenreId}/{newGenreId}")]
+        public IActionResult UpdateBookGenre(int bookId, int oldGenreId, int newGenreId)
+        {
+            if (this.library.UpdateBookGenre(bookId, oldGenreId, newGenreId))
```

Comment:
here this if with an update is looking better then in the previous place where I left the comment.
And have a chance for living as it is.